pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -498,3 +499,16 @@ func (p *DefaultProvider) ResolveClusterCIDR(ctx context.Context) error {
 	}
 	return fmt.Errorf("no CIDR found in DescribeCluster response")
 }
+
+// InjectDoNotSyncTaintsLabel adds a label for all non-custom AMI families. It is exported just for ease
+// of testing.
+// This label is to tell karpenter that it should *not* sync taints. This is to work around a race condition.
+// By default, this label is not added to the custom AMI family as users may still want their taints synced. Startup
+// taints will be racy for custom AMIs if they do not add this label, however.
+// https://github.com/kubernetes-sigs/karpenter/issues/1772
+func InjectDoNotSyncTaintsLabel(amiFamilyName string, labels map[string]string) map[string]string {
```

Comment:
Can we just inline this function? Since we won't use this function anywhere else

Reply:
We can, can't test it in isolation then though. Would it be fine to drop the unit tests?

Reply:
You can still test the CreateAMIOptions function, just like you are testing it now